src/net.h

```diff
@@ -96,9 +96,18 @@ class CConnman
         CONNECTIONS_ALL = (CONNECTIONS_IN | CONNECTIONS_OUT),
     };
 
+    struct Options
+    {
+        uint64_t nLocalServices = 0;
+        int nMaxConnections = 0;
+        int nMaxOutbound = 0;
+        int nBestHeight = 0;
+        CClientUIInterface* interface = nullptr;
```

Comment:
This will eventually hold pointers to several structures, chainparams for
example.

It may be reasonable to split it into a set of truly immutable options, and
stateful app pointers.